Fluid/NullEncoder.cs

```diff
@@ -1,77 +1,35 @@
 using System;
-using System.IO;
+using System.Runtime.CompilerServices;
 using System.Text.Encodings.Web;
 
 namespace Fluid
 {
     // An HTML encoder which passes through all input data. Does no encoding.
-    // Copied from Microsoft.AspNetCore.Razor.TagHelpers.NullHtmlEncoder.
-    public class NullEncoder : HtmlEncoder
+    public class NullEncoder : TextEncoder
     {
         private NullEncoder()
         {
         }
 
-        public static new NullEncoder Default { get; } = new NullEncoder();
+        public static NullEncoder Default { get; } = new NullEncoder();
 
-        public override int MaxOutputCharactersPerInputCharacter => 1;
-
-        public override string Encode(string value)
-        {
-            return value;
-        }
-
-        public override void Encode(TextWriter output, char[] value, int startIndex, int characterCount)
-        {
-            if (output == null)
-            {
-                throw new ArgumentNullException(nameof(output));
-            }
-
-            if (characterCount == 0)
-            {
-                return;
-            }
-
-            output.Write(value, startIndex, characterCount);
-        }
-
-        public override void Encode(TextWriter output, string value, int startIndex, int characterCount)
-        {
-            if (output == null)
-            {
-                throw new ArgumentNullException(nameof(output));
-            }
-
-            if (value == null)
-            {
-                throw new ArgumentNullException(nameof(value));
-            }
-
-            if (characterCount == 0)
-            {
-                return;
-            }
-
-            output.Write(value.Substring(startIndex, characterCount));
-        }
+        public override int MaxOutputCharactersPerInputCharacter => throw new NotImplementedException();
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public override unsafe int FindFirstCharacterToEncode(char* text, int textLength)
```

Comment:
virtual methods cannot be inlined, so attribute does little here

Reply:
https://github.com/dotnet/corefx/blob/master/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs#L89

???

Reply:
I probably need to double check with sharplab, but http://blogs.microsoft.co.il/sasha/2012/01/20/aggressive-inlining-in-the-clr-45-jit/ . Compiler intrisicts are a different thing but that example you gave is from lib, weird...

Reply:
Inlined only if sealed I have been told. I made a PR on MVC with the numbers.

Reply:
OK, I guess it's safe to inline when called method is bound to compile-time known method on a sealed class then. Got to test different JIT versions... Thanks for sharing, one always learns more 🙂